lang/ast/src/exp/typ_ctor.rs

```diff
@@ -96,7 +96,9 @@ impl Print for TypCtor {
                 fun.parens()
             }
         } else {
-            alloc.typ(&name.id).append(args.print(cfg, alloc))
+            alloc
+                .reference(&format!("{}#{}", &name.uri.to_string(), &name.id), &name.id)
```

Comment:
See the other comment regarding `Anno::Reference`. I would recommend only adding the anchor when rendering the HTML. This makes the `Reference` annotation more generically useful rather than being tied to the HTML output.